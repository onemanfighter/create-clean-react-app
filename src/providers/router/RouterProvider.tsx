import { RouterProvider as RouterProviderReactDom } from 'react-router-dom';
import { getAppRouter } from '@routes';

const RouterProvider = () => {
  // Get the auth token from the context
  return <RouterProviderReactDom router={getAppRouter(true)} />;
};

export default RouterProvider;
