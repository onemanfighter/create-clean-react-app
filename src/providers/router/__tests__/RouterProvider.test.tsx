import { render } from '@testing-library/react';
import RouterProvider from '../RouterProvider';

describe('RouterProvider', () => {
  it('should render router provider correctly', async () => {
    const { container } = render(<RouterProvider />);

    expect(container).toMatchSnapshot();
  });

  it('should render router provider correctly', async () => {
    const { container } = render(<RouterProvider />);

    expect(container).toMatchSnapshot();
  });
});
