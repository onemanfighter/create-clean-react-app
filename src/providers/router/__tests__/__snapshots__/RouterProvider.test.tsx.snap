// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouterProvider should render router provider correctly 1`] = `
<div>
  <div
    class="css-169qivt"
  >
    <div
      class="chakra-spinner css-1nkt52r"
    >
      <span
        class="css-8b45rq"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;

exports[`RouterProvider should render router provider correctly 2`] = `
<div>
  <h2>
    Unexpected Application Error!
  </h2>
  <h3
    style="font-style: italic;"
  >
    {"name":""}
  </h3>
  <p>
    ðŸ’¿ Hey developer ðŸ‘‹
  </p>
  <p>
    You can provide a way better UX than this when your app throws errors by providing your own 
    <code
      style="padding: 2px 4px; background-color: rgba(200, 200, 200, 0.5);"
    >
      ErrorBoundary
    </code>
     or
     
    <code
      style="padding: 2px 4px; background-color: rgba(200, 200, 200, 0.5);"
    >
      errorElement
    </code>
     prop on your route.
  </p>
</div>
`;
